examples/fargate/main.tf

```diff
@@ -61,7 +61,8 @@ module "eks" {
   source          = "../.."
   cluster_name    = local.cluster_name
   cluster_version = "1.20"
-  subnets         = module.vpc.private_subnets
+  subnets         = concat(module.vpc.private_subnets,module.vpc.public_subnets)
+  fargate_subnets = module.vpc.private_subnets
```

Comment:
better example will be just use as subnets `slice(module.vpc.private_subnets, 0, 2)` and for fargate `module.vpc.private_subnets[2]` or etc.

Reply:
I can change this if required, at least for the example. In your example though, whats would be the point of having different subnets for fargate and the cluster, if not to allow for external access? Wouldn't you want all the private subnets of the cluster to be used by the fargate profiles, in which case you wouldn't set fargate_subnets at all, causing it to use the subnets value for the fargate profiles anyway?

Reply:
For the purpose of the example, it is better to show usage of both `subnets` and `fargate_subnets` but for real setups, it can be different.

Reply:
I thought that motivation is to separate fargate subnets from nodes subnets (this can be common use case for some organisations which are using different standards for subnets like NACL, routing thru firewalls and etc.).

But if you really require create kubernetes service as Loadbalancer type then you dont require at all this PR.

Here you have a logic behind creating Loadbalancer type services in EKS https://docs.aws.amazon.com/eks/latest/userguide/alb-ingress.html

Reply:
That is exactly what we are using. Note that to create external lbs with the ingress controller you need to have them labeled and attached to the cluster. If they are not then the ingress controller will error saying that it has failed to find a suitable subnet. In the current module, you cannot attach public subnets to the cluster while using fargate as then it will try to attach that public subnet to the fargate profile, which is not allowed (or desirable).

Reply:
this depends probably from ingress controller which you are using. Nginx is creating just a kubernetes Loadbalancer service type and thru cloud provider create required Loadbalancer automatically.